src/style/style_layer/custom_style_layer.js

```diff
@@ -20,6 +20,11 @@ type CustomRenderMethod = (gl: WebGLRenderingContext, matrix: Array<number>) =>
  * and they should appropriately handle {@link Map.event:webglcontextlost} and
  * {@link Map.event:webglcontextrestored}.
  *
+ * The `renderingMode` property controls whether the layer is treated as `"2d"` or `"3d"` map layer. Use:
```

Comment:
very minor nit: `whether the layer is treated as **a** "2d"....`

I have a question on semantics. We don't typically talk about "3d" because as far as I know we don't really support true 3d rendering right now. Have we purposefully avoided the terminology to prevent confusion over the library's capabilities or is the difference between "3d" and something like "extruded" mostly meaningless?

Reply:
> Have we purposefully avoided the terminology to prevent confusion over the library's capabilities or is the difference between "3d" and something like "extruded" mostly meaningless?

I think it maybe helps prevent confusion when we talk about the core layer types because fill-extrusion is limited enough that saying "3d" could be misleading. But custom layers can be fully 3d. In this context it refers to whether the things drawn by the layer exist in 3d space, which is the case for both extrusions and some custom layers

Reply:
In other words, "does this layer have a meaningful Z coordinate"? That makes sense. I assumed that custom layers were flexible enough to make 3D possible, but just wanted to clarify for my own knowledge.

Reply:
Yep, that's a clearer way of putting it